reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/VariableDeclToExprReductionOpportunitiesTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.reducer.reductionopportunities;
+
+import com.graphicsfuzz.common.ast.TranslationUnit;
+import com.graphicsfuzz.common.glslversion.ShadingLanguageVersion;
+import com.graphicsfuzz.common.util.CompareAsts;
+import com.graphicsfuzz.common.util.ParseHelper;
+import com.graphicsfuzz.common.util.RandomWrapper;
+import java.util.List;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class VariableDeclToExprReductionOpportunitiesTest {
+
+  // TODO(480): Enable this test once the issue is fixed.
+  @Ignore
+  @Test
+  public void testDoNotReplace() throws Exception {
+    final String original = "void main() { int a = 1, b = 2; int c = 3; }";
+    final TranslationUnit tu = ParseHelper.parse(original);
+    final List<VariableDeclToExprReductionOpportunity> ops = VariableDeclToExprReductionOpportunities
+        .findOpportunities(MakeShaderJobFromFragmentShader.make(tu), new ReducerContext(false, ShadingLanguageVersion.ESSL_100,
+            new RandomWrapper(0), null, true));
+    assertTrue(ops.isEmpty());
+  }
+
+  @Ignore
+  @Test
+  public void testMultipleDeclarations() throws Exception {
+    final String program = "void main() {"
+        + "int a = 1;"
+        + "int b = foo();"
+        + "int c;"
+        + "}";
+    final String expected = "void main() {"
+        + " int a;"
+        + " a = 1;"
+        + " int b;"
+        + " b = foo();"
+        + " int c;"
+        + "}";
+    final TranslationUnit tu = ParseHelper.parse(program);
+    List<VariableDeclToExprReductionOpportunity> ops = VariableDeclToExprReductionOpportunities
+        .findOpportunities(MakeShaderJobFromFragmentShader.make(tu), new ReducerContext(true,
+            ShadingLanguageVersion.ESSL_100,
+            new RandomWrapper(0), null, true));
+    assertEquals(2, ops.size());
+    for (int i = 0; i < ops.size(); i++) {
```

Comment:
You could turn this into ops.forEach(VariableDeclToExprReductionOpportunity::applyReduction).  But maybe there is no point!  Up to you.

Reply:
Updated! thank you for your suggestion.